pilot/pkg/serviceregistry/kube/controller/endpointslice.go

```diff
@@ -131,8 +131,20 @@ func (esc *endpointSliceController) onEventInternal(_, ep *v1.EndpointSlice, eve
 			continue
 		}
 
+		// pure HTTP headless services should not need a full push since they do not
+		// require a Listener based on IP: https://github.com/istio/istio/issues/48207
```

Comment:
But if the port is updated from http to TCP, we still need to push lds

Reply:
wouldn't the scenario you describe be covered on by the Service update?

Reply:
You are right, already covered